solarforecastarbiter/io/api.py

```diff
@@ -834,18 +836,23 @@ def process_report_dict(self, rep_dict):
             norm = o.get('normalization')
             unc = o.get('uncertainty')
             ref_fx = o.get('reference_forecast')
+            cost_type = o.get('cost')
+            if isinstance(cost_type, dict):
+                cost = cost_type
+            else:
+                cost = costs.get(cost_type)
```

Comment:
I don't fully understand the logic. 

First case appears to cover if a cost was actually defined for the object pair. 

Does the second case cover if cost_type is None? I'd test for that explicitly like for ref_fx above and below (and I'm -1 on slicing up the ref_fx logic). Oh wait no that's not right since `cost_type` is passed onto `costs`. I guess it's a string that corresponds to a key in the costs dict. Confusing.

Reply:
yeah, I'm not even sure anymore what this does

Reply:
simplified, at least for the current structure of higher level 'costs' key